neqo-http3/src/stream_type_reader.rs

```diff
@@ -6,57 +6,120 @@
 
 #![allow(clippy::module_name_repetitions)]
 
-use crate::{AppError, Http3StreamType, HttpRecvStream, ReceiveOutput, RecvStream, Res, ResetType};
-use neqo_common::{Decoder, IncrementalDecoderUint};
+use crate::{
+    AppError, Error, Http3StreamType, HttpRecvStream, NewStreamType, ReceiveOutput, RecvStream,
+    Res, ResetType,
+};
+use neqo_common::{qtrace, Decoder, IncrementalDecoderUint};
 use neqo_transport::Connection;
 
 #[derive(Debug)]
 pub enum NewStreamTypeReader {
```

Comment:
You should write some documentation for this.  I'm not seeing how you distinguish between (unidirectional) streams that just have a type and streams that have a varint type plus a Push ID or WT Session ID.

This is also more than a type reader.  It reads a stream header.